iina/FFmpegController.m

```diff
@@ -372,4 +381,324 @@ + (NSDictionary *)probeVideoInfoForFile:(nonnull NSString *)file
   return info;
 }
 
+// MARK: - Decoding Image
+
++ (NSImage *)createNSImageWithContentsOf:(nonnull NSURL *)url
+{
+  // Variables holding objects that will need to be freed.
+  AVFormatContext *pFormatCtx = NULL;
+  AVCodecContext *pCodecCtx = NULL;
+  AVPacket *packet = NULL;
+  AVFrame *pFrame = NULL;
+  AVFrame *pFrameRGB = NULL;
+  uint8_t *pFrameRGBBuffer = NULL;
+  struct SwsContext *swsContext = NULL;
+  CGColorSpaceRef cgColorSpace = NULL;
+  CGContextRef cgContext = NULL;
+  CGImageRef cgImage = NULL;
+
+  @try {
+    char *cFilename = strdup(url.fileSystemRepresentation);
+
+#if DEBUG
+    LOG(@"Creating image with contents of file: %s", cFilename)
+#endif
```

Comment:
Consider moving this up so the strdup/free pair are closer together. I think technically the copy isn't required anyways

Reply:
Good catch. I patterned this after `getPeeksForFile` figuring it must be correct, however using `fileSystemRepresentation` directly works fine.